src/api/app/models/commit_activity.rb

```diff
@@ -0,0 +1,24 @@
+class CommitActivity < ApplicationRecord
+  belongs_to :user
+
+  validates :user, :date, :project, :package, :count, presence: true
+
+  # technically we only want entries with at least count 1, but 0s live
+  # for a brief moment during creation of the activity line
+  validates :count, numericality: { more_than_or_equal_to: 0,
+                                    only_integer: true }
+
+  def self.create_from_event_payload(payload)
+    user = User.find_by_login(payload['user'])
+    params = { user: user, date: Time.zone.today,
+               project: payload['project'], package: payload['package'] }
+    # TODO: Rails 6 supports create_or_find which is race condition free
+    begin
```

Comment:
What's wrong with `.find_or_create_by` and then increment?

Reply:
There can be [race](https://sikac.hu/use-create-or-find-by-to-avoid-race-condition-in-rails-6-0-f44fca97d16b) [conditions](https://github.com/rails/rails/pull/31989) when using find_or_create_by